mcstatus/__main__.py

```diff
@@ -120,13 +64,44 @@ def query() -> None:
             " and that the necessary port (same as server-port unless query-port is set) is open in any firewall(s)."
             "\nSee https://wiki.vg/Query for further information."
         )
-        raise click.Abort()
-    click.echo(f"host: {response.raw['hostip']}:{response.raw['hostport']}")
-    click.echo(f"software: v{response.software.version} {response.software.brand}")
-    click.echo(f"plugins: {response.software.plugins}")
-    click.echo(f'motd: "{response.motd}"')
-    click.echo(f"players: {response.players.online}/{response.players.max} {response.players.names}")
+        return
+    print(f"host: {response.raw['hostip']}:{response.raw['hostport']}")
+    print(f"software: v{response.software.version} {response.software.brand}")
+    print(f"plugins: {response.software.plugins}")
+    print(f'motd: "{response.motd}"')
+    print(f"players: {response.players.online}/{response.players.max} {response.players.names}")
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="""
+        mcstatus provides an easy way to query Minecraft servers for
+        any information they can expose. It provides three modes of
+        access: query, status, ping and json.
+        """
+    )
+
+    parser.add_argument("address", help="The address of the server.")
+
+    subparsers = parser.add_subparsers()
+    subparsers.add_parser("ping", help="Ping server for latency.").set_defaults(func=ping)
+    subparsers.add_parser(
+        "status", help="Prints server status. Supported by all Minecraft servers that are version 1.7 or higher."
+    ).set_defaults(func=status)
+    subparsers.add_parser(
+        "query", help="Prints detailed server information. Must be enabled in servers' server.properties file."
+    ).set_defaults(func=query)
+    subparsers.add_parser(
+        "json",
+        help="Prints server status and query in json." "Supported by all Minecraft servers that are version 1.7 or higher.",
+    ).set_defaults(func=json)
+
+    args = parser.parse_args()
+
+    global server
```

Comment:
I'm completely against using globals here. 

This did make sense with click, because there was pretty much no other way, however since we're calling these functions manually now (with argparse `set_defaults`, we can instead just pass in the address with `args.func(server)`. 

Globals are almost never a good practice, and they should be avoided in most cases, unless they're really the only way to do it. In this case, the use of server global also required a type ignore, which this version won't.

Reply:
Fixed in [the last force-push](https://github.com/py-mine/mcstatus/compare/a7fd2841d053ce61386f57404f12012395b5b5da..948e4e3f38ed75373cead9d8f544a89e5ccfad2e).